### ISSUE TITLE
autocomplete should escape names

### ISSUE THREAD

**Issue by TylerGrantSmith** [rstudio/rstudio#7885]
### System details

    RStudio Edition : Desktop
    RStudio Version : 1.3.959
    OS Version      : Windows 10
    R Version       :  4.0.2

### Steps to reproduce the problem
Run each assignment and tab to autocomplete
```r
x <- list("`a`" = 2)
# autocompletes to this
x$`a`
# should be this
x$`\`a\``

x <- list(`1` = 2)
# autocompletes to this
x$1
# should be this
x$`1`
```

### Describe the problem in detail
Currently it seems like the autocompletion engine won't properly escape names. 

### Describe the behavior you expected

I expected to get properly escaped names as demonstrated.  The first example results in no error, but a `NULL` object returned, and the second is a syntax error.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [ ] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [ ] If I am reporting a RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [ ] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by ronblum** [comment]
@TylerGrantSmith Thank you for raising the issue! We'll review it as we continue development of RStudio.

**Comment by kevinushey** [comment]
This actually worked fine in older versions of RStudio, so appears to be a regression. I'll take a look.

**Comment by rich-rstudio** [comment]
verified fixed on Version 1.4.938
